autoapi/templates/python/module.rst

```diff
@@ -1,19 +1,75 @@
-{{ obj.name }}
-{{ "=" * obj.name|length }}
+:mod:`{{ obj.name }}`
+======={{ "=" * obj.name|length }}
 
 .. py:module:: {{ obj.name }}
 
 {%- if obj.docstring %}
 
 .. autoapi-nested-parse::
+
    {{ obj.docstring|prepare_docstring|indent(3) }}
 
 {% endif %}
 
+{% block subpackages %}{% if obj.subpackages %}
```

Comment:
The biggest thing I think I'm currently thinking about is an option to disable the autosummary sections. They might be quite large, and we should make them optional.

Reply:
They can definitely be overkill for small packages. If we think that wanting to turn the summaries off if is a common enough use case then I think this is a good idea. It's not nice to have to copy over every template simply to turn one thing off and this would fix that.
It would be a Python only setting. I don't know if that matters.

Reply:
Seems fine to me. Maybe preprend it with `python_` or something? Feel free to push to this branch, or I can take a swing at it.

Reply:
I'd say we want default off, at least for now, on the feature as well. There are some usability issues with it and folks might not be interested in the redundancy. We can tune this as we move on though.

Reply:
I was wrong and actually autosummary does work with languages other than Python.
Unfortunately I can't push to this branch, but here's the changes I made to get this working: https://github.com/AWhetter/sphinx-autoapi/commit/59f40985c0c5e50294f72c3134aa2a5fd9a37a88

Reply:
Sorry, you should be able to push to it now.

Reply:
Can we keep PRs individual? This is confusing review a good deal.

Reply:
I'm happy to keep everything separate if that's easier. @ericholscher Shall we get your cleanup-june branch reviewed and merged first? Maybe you could include the appveyor and lint fixes in that. Then I'll fix conflicts in my branches in turn and we can merge them in order?

Reply:
The two sets of PR's conflict with each other, so that's why I merged them. I don't know a good way to review them individually when the bases conflict..

Reply:
With the changes rebased, @AWhetter has noted which commits to cherry pick on review, which can just be selected on the changes page -- ie:
https://github.com/rtfd/sphinx-autoapi/pull/116/files/010e36f15021738cd4a0e0a731fb0e6ef24519b9..782bd2c6aee84f4c9721f39d7562f14d6b43f0c4

We'll need to incrementally merge though.

+1 on addressing feedback for tests/napolean in #113 and getting that reviewed and merged.

Reply:
Those changes are already in this PR.

Reply:
We should be able to merge everything in the order that we have here using the existing PRs.

1. I'll get my fix_tests branch cleaned up and include the appveyor fixes. The tests should then all work (other than the one lint problem with `**kwargs`) and we have a good base to work on top of. We can merge this first.
2. @ericholscher, you can rebase your june-cleanup branch on top of the new master, cherry-pick any extra changes from here (Only https://github.com/rtfd/sphinx-autoapi/pull/127/commits/0cb9295b8ce0f38df05c7639da3c9a8318632082 I think?). We can review and merge that.
3. I'll rebase py_heirarchy, we'll review and merge it.
4. I'll rebase napoleon, we'll review and merge it.
5. I'll rebase basic_class_signature, we'll review and merge it.

Sound good?

Reply:
Works for me.

Reply:
Rebase & merge done. Now just need to rebase the other PR's, and we'll be ready to go :D